chainio/clients/avsregistry/writer.go

```diff
@@ -2,86 +2,225 @@ package avsregistry
 
 import (
 	"context"
+	"crypto/ecdsa"
 	"errors"
+	"math/big"
 
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
 	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
 	"github.com/Layr-Labs/eigensdk-go/chainio/txmgr"
+	"github.com/Layr-Labs/eigensdk-go/chainio/utils"
+	"github.com/Layr-Labs/eigensdk-go/crypto/bls"
 	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	gethcommon "github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/core/types"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 
-	blsoperatorstateretriever "github.com/Layr-Labs/eigensdk-go/contracts/bindings/BLSOperatorStateRetriever"
-	blspubkeyregistry "github.com/Layr-Labs/eigensdk-go/contracts/bindings/BLSPubkeyRegistry"
-	blsregistrycoordinator "github.com/Layr-Labs/eigensdk-go/contracts/bindings/BLSRegistryCoordinatorWithIndices"
+	blsapkregistry "github.com/Layr-Labs/eigensdk-go/contracts/bindings/BLSApkRegistry"
+	opstateretriever "github.com/Layr-Labs/eigensdk-go/contracts/bindings/OperatorStateRetriever"
+	regcoord "github.com/Layr-Labs/eigensdk-go/contracts/bindings/RegistryCoordinator"
 	stakeregistry "github.com/Layr-Labs/eigensdk-go/contracts/bindings/StakeRegistry"
 )
 
 type AvsRegistryWriter interface {
+	// RegisterOperatorWithAVSRegistryCoordinator is used to register a single operator with the AVS's registry coordinator.
+	//  - operatorEcdsaPrivateKey is the operator's ecdsa private key (used to sign a message to register operator in eigenlayer's delegation manager)
+	//  - operatorToAvsRegistrationSigSalt is a random salt used to prevent replay attacks
+	//  - operatorToAvsRegistrationSigExpiry is the expiry time of the signature
 	RegisterOperatorWithAVSRegistryCoordinator(
 		ctx context.Context,
+		operatorEcdsaPrivateKey *ecdsa.PrivateKey,
```

Comment:
wait why do we need ecdsaprivatekey? why not sign using signer?

Reply:
[9207cda](https://github.com/Layr-Labs/eigensdk-go/pull/91/commits/9207cdae5a927695fb78055ac785a630135b0f9c)
added comment explaining why, but basically we need to sign a msg that isnt a transaction for registering with delegation manager now, and signer can only sign transactions the way its setup.
If there's a better way to do this let me know.

Reply:
oh I see. hmm I am thinking in a case where you don't have access to private key.

Reply:
resolving for now since we added a TODO